pkg/plugins/optional/config/v1alpha/scaffolds/internal/templates/config.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import (
+	"path/filepath"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+)
+
+var _ machinery.Template = &ControllerManagerConfig{}
+
+// ControllerManagerConfig scaffolds the config file in config/manager folder.
+type ControllerManagerConfig struct {
+	machinery.TemplateMixin
+	machinery.DomainMixin
+	machinery.RepositoryMixin
+	machinery.ProjectNameMixin
+}
+
+// SetTemplateDefaults implements input.Template
+func (f *ControllerManagerConfig) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = filepath.Join("config", "manager", "controller_manager_config.yaml")
+	}
+
+	f.TemplateBody = controllerManagerConfigTemplate
+
+	f.IfExistsAction = machinery.Error
+
+	return nil
+}
+
+const controllerManagerConfigTemplate = `apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
```

Comment:
We have more files changed when the component Config flag is used.
We need to ensure that all changes made when the flag is used will be performed when we use the plugin to init a project OR to edit a project that exist already.

Reply:
@camilamacedo86 - Yes, right. In this PR, I just scaffolded one file. There are a couple of files which has the `--leader-elect` flag enabled according to the flag. I will update those too in the PR.